src/routes/admin/kitchen/+page.svelte

```diff
@@ -0,0 +1,12 @@
+<script lang="ts">
+  import OrderList from "./OrderList.svelte";
+  import { orders } from "$lib/stores/orderStore";
+</script>
+
+<div class="grid h-full grid-cols-[3fr,auto,1.5fr,auto,1fr] grid-rows-[100%] gap-4">
+  <OrderList {orders} show={["received"]} state={"production"} />
+
+  <div class="divider divider-horizontal"></div>
+
+  <OrderList {orders} show={["production"]} state={"completed"} />
```

Comment:
Vil du også gjøre refaktoriseringen for OrderList i denne PR? Jeg bistår gjerne!

Jeg bare deler litt tanker her angående min erfaring med `OrderList, føl deg fri til å ignorere noe eller alt.

Kan vi med fordel gjøre `OrderList` om til:

```
<each $orders as order>
  <Order order onclick={(order) => order.setState('completed')} >
 </each>
 ```
 
Altså, å flytte iterasjonen opp ett nivå? Tailwind for listens spacing og slikt blir duplisert. Vi kan ekstrahere dette som CSS, eller et `List` komponent.
 
 Om ikke, er en bedre API for `OrderList ` i mitt hode:
 
 ```
 <script>
const  completedOrders = $orders.filter(o => o.state === 'completed')
 </script>
 <OrderList orders={completedOrders} onclick={() => orders.setState(order.id, state)} />
 ```

Reply:
Jeg likte veldig godt å filter _før_ orders blir passet. Når jeg gjør det så får jeg at `orders is not a store with a subscribe method`. orderStore har jo det implementert, men jeg aner ikke hvorfor det ikke virker.

Å bruke onclick utenfor componenten slik vil ikke virke for OrderList da den ikke vet hva order er utenfor componenten.

Reply:
@IldenH jeg husker å få denne feilmeldingen da jeg skrev modulen. Hvis man autosubscriber med `$store`, så kan man få denne feilmeldingen. Derfor tror eg at i det tilfellet du beskriver, er orders sendt ned som en store hvilket er autosubscribet, alas `$orders`, for å så brukes `$orders` på enda en gang.

Bedre typing burde ha hjulpet.

Du har rett med at slik det ser ut nå vil det ikke funke. Men man kan sende den spesifikke ID eller order til calbacken, jeg glemte bare å legge til parameteret. Fikset.